### PR TITLE
feat: add ajv

### PR DESCRIPTION
**Pull request by rehanvdm**
Adds AJV to test cases, I saw the open ticket #683 and agree that AJV is a very popular package that is used in a lot of places. 

AJV now has something called AJV standalone mode. 
>Ajv supports generating standalone validation functions from the JSON Schema into functions at compile/build
time. These functions can then just be consumed during runtime to do validation checks against JSON.

You can read more about it here https://ajv.js.org/standalone.html and you can also see some more examples and documentation in this PR I made https://github.com/ajv-validator/ajv/pull/1866. What I am getting at is that you can use other packages to map TS native types to JSON Schema, then from that create validation functions for your types at compile time which are consumed at runtime. Throw this in a `watch` command and it makes for a great DX as well. (I have a blog that I will publish as soon as that PR is merged about this method).

AJV is actually the fastest, below is a screenshot of the generated CSV file 
image

I saw the discussion here https://github.com/moltar/typescript-runtime-type-benchmarks/issues/497 and can only mention that one pain point is the Date object. As soon as you have that in your object, you have to first `JSON.parse(JSON.stringify(...))` the data to force all objects to be represented as their "primitive" types, which for the Date object is an ISO 8601 formatted string. You then have to add the format string property and additional ajv formats to make that work. So again difficult to compare apples and apples. Just thought I mention it. But for the specific test data and schema AJV is the fastest at the moment.

### CODE REVIEW

File: cases/ajv.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import Ajv from "ajv"
+import { Case } from './abstract';
+
+const ajv = new Ajv();
```
Comment (moltar) [inline cases/ajv.ts:4]: We seem to have lots of lint errors:

```
 Error:    1:17  error    Replace `"ajv"` with `'ajv';`                                                                         prettier/prettier
Warning:    1:17  warning  Strings must use singlequote                                                                          quotes
Error:    6:8   error    Replace `"AjvTest"` with `'AjvTest'`                                                                  prettier/prettier
```

Please run lint fix :)

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "@toi/toi": "1.3.0",
     "@typeofweb/schema": "0.7.3",
     "@types/benchmark": "2.1.1",
+    "ajv": "^6.12.6",
```
Comment (moltar) [inline package.json:26]: Please use a fixed version.

Renovate will keep them up-to-date.

**Review by moltar — CHANGES_REQUESTED** [review]
Thanks for the PR.

Didn't know about the AJV compiled mode. Looks great! Will have to play around with it myself.

A few small changes, if you don't mind :)

File: cases/ajv.ts (hunk 1 above)
Comment (rehanvdm) [reply cases/ajv.ts]: Done, sorry, I am used to husky git hooks doing its thing before pushing :)

File: package.json (hunk 2 above)
Comment (rehanvdm) [reply package.json]: I didn't even notice that, done.

**Comment by rehanvdm** [comment]
I am not sure why it installed version 6 of AJV as the latest is now 8. After installing it, I ran the tests again, but got weirdly inconsistent results over all of the packages, not sure why might be the thermal protection kicking in, throttling the CPU voltage 🤷‍♂️. Let's see what it produces on the build machine.

**Comment by rehanvdm** [comment]
I also noticed npm upgraded the package lock to V2, which I hope isn't a problem?

**Comment by moltar** [comment]
> I also noticed npm upgraded the package lock to V2, which I hope isn't a problem?

Hope so too 😂

**Pull request merged**